src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+        case "todo":
+        case "deadline":
+        case "event":
+            throw new DukeException("Please enter a valid"
+                    + " description for your task!");
+        case "done":
+            throw new DukeException("Please enter the ID "
+                    + "of the task you would like to complete.");
+        case "delete":
+            throw new DukeException("Please retry and enter "
+                    + "the ID of the task to be deleted.");
+        case "check":
+            throw new DukeException("Please enter a date to check!");
+        case "find":
+            throw new DukeException("Please enter a keyword to search for.");
+        default:
+            throw new DukeException("Please enter a valid "
+                    + "command into the console.");
+        }
+    }
+
+    /**
+     * Parses two word commands. (e.g. done 2, delete 3)
+     * @param modEcho The command entered by the user.
+     * @return A command.
+     * @throws DukeException If the command entered is invalid.
+     */
+    public Command checkTwoWords(String[] modEcho) throws DukeException {
+        String task = modEcho[0];
+
+        switch (task) {
+        case "done":
+            if (!isNumeric(modEcho[1])) {
+                throw new DukeException("Please enter a valid task number to complete!");
+            } else {
+                int index = Integer.parseInt(modEcho[1]) - 1;
+                return new DoneCommand(index);
+            }
+
+        case "delete":
+            if (!isNumeric(modEcho[1])) {
+                throw new DukeException("Please enter a valid task number for deletion!");
+            } else {
+                int index = Integer.parseInt(modEcho[1]) - 1;
+                return new DeleteCommand(index);
+            }
+
+        case "check":
+            try {
+                LocalDate checkedDate = LocalDate.parse(modEcho[1]);
+                return new CheckCommand(checkedDate);
+            } catch (DateTimeParseException ex) {
+                throw new DukeException("Please enter the date in this format: yyyy-mm-dd");
+            }
+
+        case "find":
+            return new FindCommand(modEcho[1]);
+        default:
+            if (task.equals("todo")) {
+                return new TodoCommand(modEcho[1]);
+            } else if (task.equals("deadline") || task.equals("event")) {
+                return checkValidTime(modEcho);
+            } else {
+                throw new DukeException("Please enter a valid task name to add into the list!");
+            }
+
+        }
+    }
+
+    /**
+     * Parses commands involving time-based tasks.
+     * @param modEcho The command entered by the user.
+     * @return A command.
+     * @throws DukeException If the date entered is in the wrong format.
+     */
+    public Command checkValidTime(String[] modEcho) throws DukeException {
+        String task = modEcho[0];
+        String[] processTime = modEcho[1].split("/");
```

Comment:
I feel that the processTime name can be better? Maybe it can be called as processTimePrefix?